airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -97,59 +95,51 @@ def __post_init__(self, parameters: Mapping[str, Any]):
     def get_stream_state(self) -> StreamState:
         return {self.cursor_field.eval(self.config): self._cursor} if self._cursor else {}
 
-    def update_cursor(self, stream_slice: StreamSlice, last_record: Optional[Record] = None):
+    def set_initial_state(self, stream_state: StreamState) -> None:
+        """
+        Cursors are not initialized with their state. As state is needed in order to function properly, this method should be called
+        before calling anything else
+
+        :param stream_state: The state of the stream as returned by get_stream_state
+        """
+        self._cursor = stream_state.get(self.cursor_field.eval(self.config)) if stream_state else None
```

Comment:
I don't have any strong opinion. If we were to do this:

### Benefits
* It would prevent weird error where the state initialized multiple time (behavior that we assume is the current one)

### Drawbacks
* This would change the actual behavior
* This would need to be done on every implementations

Personally, I think the real solution is that any cursor receives their state as part of `__init__` but this is very backward to the way this is done today and would imply changing the entrypoint and therefore probably affecting all the sources unfortunately...

Reply:
Got it, given that the current behavior doesn't do this check I'm fine not adding it in here.